contents/tutorials/remix-ab-tests.md

```diff
@@ -0,0 +1,295 @@
+
+Replace your code in `_index.tsx` with the following:
+
+```ts file=_index.tsx
+import { LoaderFunctionArgs , json } from "@remix-run/node";
+import { useLoaderData } from '@remix-run/react';
+import posthog from "posthog-js";
+import PostHogNodeClient from "../posthog"
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const posthogNode = PostHogNodeClient()
+  let buttonText = 'No variant'
+  try {
+    const distinctId = 'placeholder-user-id'
+    const enabledVariant = await posthogNode.getFeatureFlag('my-cool-experiment', distinctId);
+    if (enabledVariant === 'control') {
+      buttonText = 'Control Variant';
+    } else if (enabledVariant === 'test') {
+      buttonText = 'Test Variant';
+    }
+  } catch (error) {
+    buttonText = 'Error';
+  }
+
+  return json({ buttonText });
+}
+
+export default function Index() {
+  const data = useLoaderData<typeof loader>();
+
+  const handleClick = () => {
+    posthog.capture('home_button_clicked');
+  };
+
+  return (
+    <div id="app">
+      <h1>Remix A/B Test</h1>
+      <button onClick={handleClick}>{data.buttonText}</button>
+    </div>
+  );
+}
+```
+
+Now, when you refresh the page, the button text is already set when the page loads. 
+
+#### Setting the correct `distinctId`
+
+You may notice that we set `distinctId = 'placeholder-user-id'` in our flag call above. In production apps, to ensure you fetch the correct flag value for your user, `distinctId` should be set to their unique ID. 
+
+For logged-in users, you typically use their email as their `distinctId`. However, for logged-out users, you can use the `distinct_id` property from their PostHog cookie:
+
+```ts file=_index.tsx
+import { LoaderFunctionArgs , json } from "@remix-run/node";
+import { useLoaderData } from '@remix-run/react';
+import posthog from "posthog-js";
+import PostHogNodeClient from "../posthog"
+import { parse } from 'cookie';
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const posthogNode = PostHogNodeClient()
+  let buttonText = 'No variant'
+
+  const projectAPIKey = '<ph_project_api_key>' // your PostHog API key from your App Settings
+  const cookies = parse(request.headers.get('cookie') || '');
+  const cookieKey = `ph_${projectAPIKey}_posthog`;
+  if (cookies[cookieKey]) {
+    try {
+      const distinctId = JSON.parse(cookies[cookieKey]).distinct_id;
+      const enabledVariant = await posthogNode.getFeatureFlag('my-cool-experiment', distinctId);
+      if (enabledVariant === 'control') {
+        buttonText = 'Control Variant';
+      } else if (enabledVariant === 'test') {
+        buttonText = 'Test Variant';
+      }
+    } catch (error) {
+      buttonText = 'Error';
+    }
+  }
+
+  return json({ buttonText });
+}
+
+// rest of your code
+```
+
```

Comment:
Should we add a note here that they need to have a fallback (like using the client side) if the cookie doesn't exist? (I know I have suggested every possible solution to this problem across all these tutorials 😅 , but this might be the easiest one).

Reply:
Added a note!